chore: Add provisional docs for auth usage (#109)
netlify/functions: README.md

- `HandlerEvent`
- `HandlerResponse`

## Netlify Integration Auth Management

> To make building on and interacting with third-party APIs as simple and powerful as possible, Netlify provides API secret provisioning and management, powered by OneGraph. It’s enabled on a per-site basis under the [Netlify labs](https://app.netlify.com/user/labs) tab, where you can visually select which services you want to make available for your functions or site builds, and which scopes you need access to.

Reviewer: Sounds like this location may shift (enable in Netlify Labs, configure elsewhere in the UI)

Reviewer 2: I believe the plan is to get it live while the UI lives inside of the Labs UI, so the readme should be updated to reflect this and then updated again later once we undertake the review

### Usage

> After you’ve enabled one or more services, you can access the relevant API tokens and secrets in your functions with the `getSecrets` function exported from this package:

Reviewer: A couple of small suggestions for clarity:

```suggestion
After you’ve enabled one or more services, you can access the relevant API tokens and secrets in your serverless functions with the `getSecrets` function exported from the `@netlify/functions` package.
```

      // Handle all fetching, refreshing, rotating, etc. of tokens
      const secrets = await getSecrets();
    
      // Check if the GitHub auth has been enabled for the site
      if (!secrets.gitHub?.bearerToken) {
        return {
          statusCode: 412,
          body: JSON.stringify({
            error: "You must enable the GitHub auth in your Netlify dashboard",
          }),
          headers: {
            "Content-Type": "application/json",
          },
        };
      }
    
      // If so, we can make calls to the GitHub API - REST or GraphQL!
      const MyOctokit = Octokit.plugin(restEndpointMethods);
      const octokit = new MyOctokit({ auth: secrets.gitHub.bearerToken });
    
      // We'll list all open issues on the netlify/functions repository
      const result = await octokit.rest.issues.list({
        owner: "netlify",
        repo: "functions",
        state: "open",
      });
    
      return {
        statusCode: 200,
        body: JSON.stringify(result),
        headers: {
          "Content-Type": "application/json",
        },
      };
    };
  ```

### Checking additional metadata about auths in your functions

> Netlify Integration Auth Management also tracks metadata for the installed auth tokens. If you would like to verify that an auth has been installed with the correct scopes before calling into an API (say, for example, to give a better error message in the developer logs):

Reviewer: Minor edit to intro:
```suggestion
Netlify Integration Auth Management also tracks metadata for installed auth tokens. You can verify that an auth has been installed with the correct scopes before calling into an API (say, for example, to give a better error message in the developer logs). Here's an example:
```

    
      // Secrets have an optional `grantedScopes` field that has details
      // on what the auth is allowed to do
      const tokenHasScope = secrets.gitHub.grantedScopes?.some((grantedScope) =>
        sufficientScopes.includes(grantedScope.scope)
      );
    
      // Notice how we can leave a great error message that tells us exactly what
      // we need to do to fix things.
      if (!tokenHasScope) {
        return {
          statusCode: 412,
          body: JSON.stringify({
            error: `You have enabled GitHub auth in your Netlify Auth dashboard, but it's missing a required scope. The auth must have one (or both) of the scopes: ${sufficientScopes.join(", ")}`,
          }),
          headers: {
            "Content-Type": "application/json",
          },
        };
      }
    
     // ...
    }
  ```

### Accessing integration auth during development

When running your site under `netlify dev`, the environmental variables that power the auth management will be synced, and you can transparently develop against the third party APIs - no additional configuration necessary!

### Updating or removing auth from your site

> At any time you can revisit the Netlify Integration Auth Management tab for your site to see the installed auth. From there, you can select new scopes for already-installed auth and then run through the browser-based auth flow again, and the new scopes will be available to all your existing, deployed functions instantly. 

Reviewer: This would benefit from being more specific about where the user can update items in the UI. (The location may change, however.)

Author: Included a link and instructions for the current location

> You can also install new services, or remove currently-installed services you’re no longer using.

Reviewer: Don't need this comma
```suggestion
You can also install new services or remove currently-installed services you’re no longer using.
```
